mimic/model/nova_objects.py

```diff
@@ -798,6 +798,23 @@ def request_delete(self, http_delete_request, server_id):
         server.update_status(u"DELETED")
         return b''
 
+    def request_action(self, http_action_request, server_id):
+        """
+        Perform the requested action on the provided server
+        """
+        server = self.server_by_id(server_id)
+        if server is None:
+            return dumps(not_found("Instance " + server_id + " could not be found",
+                                   http_action_request))
+        action_json = loads(http_action_request.content.read())
+        if 'resize' in action_json:
+            flavor = action_json['resize']['flavorRef']
```

Comment:
If flavorRef is not provided, this will fail with a keyerror. (I had made the same comment earlier, guess it got lost due to other changes)

Can we handle that scenario as follows:
```
flavor = action_json['resize'].get('flavorRef')
if not flavor:
    return dumps(bad_request("Resize requests require 'flavorRef' attribute.", http_action_request))
```